aubo_i5:
  # publish all joint states--------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Gripper controller
  gripper_controller:
    type: position_controllers/GripperActionController
    joint: robotiq_85_left_knuckle_joint
    action_monitor_rate: 20
    goal_tolerance: 0.002
    max_effort: 100
    stall_velocity_threshold: 0.001
    stall_timeout: 1.0

  # arm controller
  arm_controller:
    type: position_controllers/JointTrajectoryController
    joints:
      - shoulder_joint
      - upperArm_joint
      - foreArm_joint
      - wrist1_joint
      - wrist2_joint
      - wrist3_joint
    constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      shoulder_joint: {trajectory: 0.1, goal: 0.1}
      upperArm_joint: {trajectory: 0.1, goal: 0.1}
      foreArm_joint: {trajectory: 0.1, goal: 0.1}
      wrist1_joint: {trajectory: 0.1, goal: 0.1}
      wrist2_joint: {trajectory: 0.1, goal: 0.1}
      wrist3_joint: {trajectory: 0.1, goal: 0.1}

    stop_trajectory_duration: 0.5
    state_publish_rate: 25
    action_monitor_rate: 10

    # gains:
    #   shoulder_joint:   {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}
    #   upperArm_joint: {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}
    #   foreArm_joint:  {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}
    #   wrist1_joint:       {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}
    #   wrist2_joint:    {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}
    #   wrist3_joint:      {p: 1000.0, i: 0.0, d: 0.1, i_clamp: 0.0}



  joint_group_position_controller:
    type: position_controllers/JointGroupPositionController
    joints:
      - shoulder_joint
      - upperArm_joint
      - foreArm_joint
      - wrist1_joint
      - wrist2_joint
      - wrist3_joint
      - robotiq_85_left_knuckle_joint